flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -236,21 +237,23 @@ protected ResultPartitionWriter createResultPartition(
 			NetworkEnvironment environment,
 			int channels) throws Exception {
 
-		ResultPartition resultPartition = new ResultPartition(
+		NetworkResultPartition networkResultPartition = new NetworkResultPartition(
 			"sender task",
-			new NoOpTaskActions(),
-			jobId,
 			partitionId,
 			ResultPartitionType.PIPELINED_BOUNDED,
 			channels,
 			1,
 			environment.getResultPartitionManager(),
+			ioManager);
+		environment.setupPartition(networkResultPartition);
+
+		ResultPartition resultPartition = new ResultPartition(
```

Comment:
if this `resultPartition` has `NoOpTaskActions` and `NoOpResultPartitionConsumableNotifier`, do we still need to wrap `networkResultPartition` with it? Could we just return `networkResultPartition`?

Reply:
From functional aspect, I agree with your point.  But it seems reasonable to reflect the real path for performance sensitive benchmark. If there are extra overheads executed in `ResultPartition`, it could be detected in the benchmark. What do you think?

Reply:
ok, let's keep it